proposals/0033-timely-vote-credits.md

```diff
@@ -0,0 +1,389 @@
+that gives all validators a "reasonable" amount of time to land their votes.
+This reduces the credits earning disadvantage that comes from being more
+distant from the network.  A balance has been struck between the strictest
+rewards schedule, which most strongly discourages intentional lagging, and
+more lenient rewards schedules, which improves credit earnings for distant
+validators who are not artificially lagging.
+
+Historical voting data has been analyzed over many epochs and the data shows
+that the smallest grace period that allows for very minimal impact on well
+behaved distant validators is 2 slots, which means that all slots voted on
+within 2 slots will award maximum vote credits to the voting validator.
+This gives validators about 1 second to land their votes without penalty.
+The maximum latency between two points on Earth is about 100 ms, so allowing
+a full 1,000 ms latency without penalty does not have adverse impact on
+distant validators.
+
+Another factor to consider is what the maximum vote credits to award for a
+vote should be.  Assuming linear reduction in vote credits awarded (where 1
+slot of additional lag reduces earned vote credits by 1), the maximum vote
+credits value determines how much "penalty" there is for each additional
+slot of latency.  For example, a value of 10 would mean that after the grace
+period slots, every additional slot of latency would result in a 10%
+reduction in vote credits earned as each subsequent slot earns 1 credit less
+out of a maximum possible 10 credits.
+
+Again, historical voting data was analyzed over many epochs and the
+conclusion drawn was that a maximum credits of 8 is the largest value that
+can be used and still have a noticeable effect on known laggers.  Values
+higher than that result in such a small penalty for each slot of lagging
+that intentional lagging is still too profitable.  Lower values are even
+more punishing to intentional lagging; but an attempt has been made to
+conservatively choose the highest value that produces noticeable results.
+
+When a Vote or VoteStateUpdate instruction is received by a validator, it
+uses the Clock sysvar to identify the slot in which that instruction has
+landed.  For any newly voted on slot within that Vote or VoteStateUpdate
+transaction, the validator will record the vote latency of that slot as
+(voted_in_slot - voted_on_slot).
+
+These vote latencies are stored in u8 field of the Lockout data structure
+named *latency*.
+
+The credits awarded for a rooted slot are calculated using the latency value
+stored in the latency field the slot, using a formula that awards latencies
+of 1 - 2 slots 8 credits, with a 1 credit reduction for each vote latency
+after 2.  Rooted slots will always be awarded a minimum credit of 1 (never
+0) so that very old votes, possibly necessary in times of network stress,
+are not discouraged.
+
+The implementation of the above is fairly simple for Vote transactions,
```

Comment:
The remaining part of this section is not needed as `rooted slot` is well defined for both Vote and VoteStateUpdate instructions and was designed as part of the initial specs/proposal. Just makes it more complex re-explaining the rooting mechanism already present in the vote program.

Reply:
Also it allows me to refer to existing implementation to give assurance that the melding of the new implementation into the old won't cause problems, and to describe why that is true.  But again I will remove it.